src/stores/ui.ts

```diff
@@ -1,33 +1,148 @@
   /**
    * whether to show the instructions snackbar.
    */
   instructionsSnackbarState: SnackbarState
+  /**
+   * whether the filters are shown (sidebar on desktop or modal on mobile).
+   */
+  isFilterVisible: boolean
+  /**
+   * (only for desktop layout) whether the filters were dismissed. If true,
+   * the filters should be closed on SSR for the first load on desktop layout.
+   */
+  isFilterDismissed: boolean
+  /**
+   * whether the site layout is desktop (or mobile).
+   */
+  isDesktopLayout: boolean
+  /**
+   * whether the request user agent is mobile or not.
+   */
+  isMobileUa: boolean
 }
 
 export const useUiStore = defineStore('ui', {
   state: (): UiState => ({
-    instructionsSnackbarState: 'not_shown',
+    instructionsSnackbarState: NOT_SHOWN,
+    isFilterVisible: false,
+    isFilterDismissed: false,
+    isDesktopLayout: false,
+    isMobileUa: true,
   }),
 
   getters: {
-    areInstructionsVisible(state) {
-      return state.instructionsSnackbarState === 'visible'
+    areInstructionsVisible(state): boolean {
+      return state.instructionsSnackbarState === VISIBLE
     },
   },
 
   actions: {
     showInstructionsSnackbar() {
-      if (this.instructionsSnackbarState === 'not_shown') {
-        this.instructionsSnackbarState = 'visible'
+      if (this.instructionsSnackbarState === NOT_SHOWN) {
+        this.instructionsSnackbarState = VISIBLE
       }
     },
     hideInstructionsSnackbar() {
-      this.instructionsSnackbarState = 'dismissed'
+      this.instructionsSnackbarState = DISMISSED
+    },
+    /**
+     * Given a list of key value pairs of UI state parameters and their states,
+     * populate the store state to match the cookie.
+     *
+     * Values stored in the cookie are stored across sessions and are modified
+     * if the breakpoints change.
+     *
+     * @param cookies - mapping of UI state parameters and their states.
+     */
+    initFromCookies(cookies: UiStateCookie) {
+      this.isDesktopLayout = cookies.isDesktopLayout ?? false
+      this.isMobileUa = cookies.isMobileUa ?? false
+      this.isFilterDismissed = cookies.isFilterDismissed ?? false
+      this.isFilterVisible = this.isDesktopLayout
+        ? !this.isFilterDismissed
+        : false
+    },
+    /**
+     * Sets the breakpoint and ua state. We have to set the cookie value inside
+     * the `setup` function because `app.$cookies` is not available outside it.
+     * This is why we return the object that needs to be saved in the cookies.
+     * @param isDesktopLayout - whether the layout is desktop (`lg` with the `new_header`
+     * and `md` with the `old_header`).
+     * @param isMobileUa - whether the UA is `mobile` or not.
+     */
+    updateBreakpoint(
+      isDesktopLayout: boolean,
+      isMobileUa: boolean | undefined
+    ): ScreenParameters | undefined {
+      if (
+        this.isDesktopLayout === isDesktopLayout &&
+        typeof isMobileUa === 'undefined'
+      )
+        return
```

Comment:
What does this condition mean? Can you add a comment explaining it? Also worth documenting: why is `isMobileUa` allowed to be undefined?